src/providers/swagger/swagger.provider.ts

```diff
@@ -0,0 +1,256 @@
+
+  const Query = (input as any).Query;
+
+  // derive query parameters from io-ts definition of e.Input.Query
+  const queryParameters =
+    Query !== undefined
+      ? R.keys(Query.props).reduce<any[]>(
+          (acc, k) =>
+            acc.concat({
+              name: k,
+              in: 'query',
+              required: Query?._tag === 'PartialType',
+              schema: getOpenAPISchema((Query?.props)[k]),
+            }),
+          []
+        )
+      : [];
+
+  // console.log('endpoint query', e.Input.Params);
+
+  const Params = (input as any).Params;
+  // derive path parameters from io-ts definition of e.Input.Params
+  const pathParameters =
+    Params !== undefined
+      ? R.keys(Params.props).reduce<any[]>(
+          (acc, k) =>
+            acc.concat({
+              name: k,
+              in: 'path',
+              required: true,
+              schema: getOpenAPISchema((Params?.props)[k]),
+            }),
+          []
+        )
+      : [];
+
+  // combine all parameters
+  const parameters = [
+    ...headerParameters,
+    ...queryParameters,
+    ...pathParameters,
+  ];
+  //   console.log('parameters', parameters);
+
+  // add definition of request body, if needed
+  const requestBody = hasRequestBody(e)
+    ? {
+        content: {
+          'application/json': {
+            schema: getOpenAPISchema((input as any)?.Body),
+          },
+        },
+      }
+    : {};
+
+  // define success response
+  const successResponse = {
+    [responseStatusCode]: {
+      description: (e.Output as any).name,
+      content: {
+        'application/json': {
+          schema: getOpenAPISchema(e.Output as any),
+        },
+      },
+    },
+  };
+
+  // TODO: define error response
+
+  return {
+    summary: key,
+    description: `${e.Method} ${path}`,
+    tags: tags,
+    parameters,
+    security,
+    requestBody,
+    responses: {
+      ...successResponse,
+    },
+  };
+};
+
+export const generateDoc = (config: DocConfig): any => {
+  const paths = pipe(
+    config.endpoints,
+    R.reduceWithIndex({}, (versionKey, acc, versionEndpoints) => {
+      return pipe(
+        versionEndpoints,
+        R.reduceWithIndex(acc, (scopeKey, scopeAcc, scopeEndpoints) => {
+          return pipe(
+            scopeEndpoints,
+            R.reduceWithIndex(S.Ord)({}, (key, endpointAcc, endpoint) => {
+              // get swagger compatible path
+              const endpointStaticPath = endpoint.getStaticPath(
+                (param) => `{${param}}`
+              );
+              const prevEndpoints =
+                (endpointAcc as any)[endpointStaticPath] ?? undefined;
+
+              const previousMethodSchema =
+                prevEndpoints?.[endpoint.Method.toLowerCase()];
+              const currentEndpointSchema = apiSchemaFromEndpoint(
+                key,
+                endpoint,
+                [`${versionKey} - ${scopeKey}`]
```

Comment:
this is the label for grouping the api by `version` and `scope` (public/creator)